### ISSUE TITLE
Eigenvalues of a general matrix

### ISSUE THREAD

**Issue by lpawela** [josdejong/mathjs#2439]
I'm trying to visualize the numerical range of a matrix on a website. For this, I need to calculate the eigenvalues of a general matrix with complex elements. This should be available since #1743 was merge. Yet, I get the following errors.

```html
<script src="https://unpkg.com/mathjs@10.1.1/lib/browser/math.js"></script>

<script type="text/javascript">
console.log(math.eigs([[1, 2], [3, 4]]));
console.log(math.eigs([[3, -2], [4, -1]]));
console.log(math.eigs([[3, -2], [math.complex(4, 2), -1]]));
</script>
```

The first two line work as expected, the last one results in the following error
```
Uncaught TypeError: No ordering relation is defined for complex numbers
    at Function.Complex, Complex (smaller.js:75:13)
    at smaller (typed-function.js:1086:85)
    at complexEigs.js:111:18
    at complexEigs.js:22:15
    at j (eigs.js:100:12)
    at Function.Array (eigs.js:51:14)
    at Object.eigs (typed-function.js:1085:85)
    at (index):1:14924
Complex, Complex	@	smaller.js:75
smaller	@	typed-function.js:1086
(anonymous)	@	complexEigs.js:111
(anonymous)	@	complexEigs.js:22
j	@	eigs.js:100
Array	@	eigs.js:51
eigs	@	typed-function.js:1085
(anonymous)	@	(index):1
```

Am I doing something? Did I misunderstand the nature of #1743?

**Comment by gwhitney** [comment]
This is definitely a bug. Thank you so much for reporting! Someone will look at it as soon as they are able, but as you can see, there is a big backlog; any suggestions as to how to address it or better yet a PR fixing it would definitely be welcome.

**Comment by lpawela** [comment]
@gwhitney playing around with the console in my browser I think I found the issue. These lines https://github.com/josdejong/mathjs/blob/38d9466e2caf9935724afdc5180abc6869155ea0/src/function/matrix/eigs/complexEigs.js#L67-L68 are defined based on the type of the input. In the case I found they are complex numbers. Later, we have https://github.com/josdejong/mathjs/blob/38d9466e2caf9935724afdc5180abc6869155ea0/src/function/matrix/eigs/complexEigs.js#L90-L91 which makes both norms complex. Next, we add to the norms the absolute values of matrix elements https://github.com/josdejong/mathjs/blob/38d9466e2caf9935724afdc5180abc6869155ea0/src/function/matrix/eigs/complexEigs.js#L96-L97 which keeps the norms complex. This in turn makes the variables https://github.com/josdejong/mathjs/blob/38d9466e2caf9935724afdc5180abc6869155ea0/src/function/matrix/eigs/complexEigs.js#L108-L109 complex as well. Finally, we hit the offending line https://github.com/josdejong/mathjs/blob/38d9466e2caf9935724afdc5180abc6869155ea0/src/function/matrix/eigs/complexEigs.js#L111 which results in a comparison between a real `c` and a complex `rowDivRadix`. I don't know much about javascript, and I don't have a proper environment to prepare and submit a PR, but my suggestion would be to change `zero` and `one` to a real value. Would that work, or will it break something?

**Comment by gwhitney** [comment]
By Jove that's it. All of the norms and radices in this part of the algorithm are intended to be real even in the case of complex entries. Fixing the types prevented the error from being thrown, and produced the correct eigenvalues, Hopefully we can get the PR above merged reasonably quickly and a new release to fix this serious bug. Thank you so much Łukasz for your help. And @josdejong let me know if there's anything in particular I should do to expedite this fix.

**Comment by josdejong** [comment]
wow 😳 , thanks for debugging this @lpawela and @gwhitney for working it out in a fix. Will publish the fix today or tomorrow

**Comment by josdejong** [comment]
The fix is published now in `v10.3.0`
